tests/pytests/unit/roster/test_dir.py

```diff
@@ -0,0 +1,145 @@
+
+    opts = salt.config.master_config(os.path.join(RUNTIME_VARS.TMP_CONF_DIR, "master"))
+    utils = salt.loader.utils(opts, whitelist=["json", "stringutils", "roster_matcher"])
+    runner = salt.loader.runner(opts, utils=utils, whitelist=["salt"])
+    return {
+        dir_: {
+            "__opts__": {
+                "extension_modules": "",
+                "optimization_order": [0, 1, 2],
+                "renderer": "jinja|yaml",
+                "renderer_blacklist": [],
+                "renderer_whitelist": [],
+                "roster_dir": os.path.join(TESTS_DIR, "unit/files/rosters/dir"),
+                "roster_domain": roster_domain,
+            },
+            "__runner__": runner,
+            "__utils__": utils,
+        }
+    }
+
+
+def _test_match(ret, expected):
+    """
+    assertDictEquals is too strict with OrderedDicts. The order isn't crucial
+    for roster entries, so we test that they contain the expected members directly.
+    """
+    assert ret != {}, "Found no matches, expected {}".format(expected)
+    for minion, data in ret.items():
+        assert minion in expected, "Expected minion {} to match, but it did not".format(
+            minion
+        )
+        assert (
+            dict(data) == expected[minion]
+        ), "Data for minion {} did not match expectations".format(minion)
+
+
+def test_basic_glob(expected):
+    """Test that minion files in the directory roster match and render."""
+    expected = expected["basic"]
+    ret = dir_.targets("*_basic", saltenv="")
+    _test_match(ret, expected)
+
+
+def test_basic_re(expected):
+    """Test that minion files in the directory roster match and render."""
+    expected = expected["basic"]
+    ret = dir_.targets(".*basic$", "pcre", saltenv="")
+    _test_match(ret, expected)
+
+
+def test_basic_list(expected):
+    """Test that minion files in the directory roster match and render."""
+    expected = expected["basic"]
+    ret = dir_.targets(expected.keys(), "list", saltenv="")
+    _test_match(ret, expected)
+
+
+def test_roster_domain(expected):
+    """Test that when roster_domain is configured, it will provide a default hostname
+    in the roster of {filename}.{roster_domain}, so that users can use the minion
+    id as the local hostname without having to supply the fqdn everywhere."""
+    expected = expected["domain"]
+    ret = dir_.targets(expected.keys(), "list", saltenv="")
+    _test_match(ret, expected)
+
+
+def test_empty(expected):
+    """Test that an empty roster file matches its hostname"""
+    expected = expected["empty"]
+    ret = dir_.targets("*_empty", saltenv="")
+    _test_match(ret, expected)
+
+
+def test_nomatch():
+    """Test that no errors happen when no files match"""
+    try:
+        ret = dir_.targets("", saltenv="")
+    except:
+        pytest.fail(
+            "No files matched, which is OK, but we raised an exception and we should not have."
+        )
+        raise
```

Comment:
```suggestion
    except:
        pytest.fail(
            "No files matched, which is OK, but we raised an exception and we should not have."
        )
```
We won't reach this line.
And, if we shouldn't have exceptions we should catch them and allow the test to fail based on that exception being raised...